### PR TITLE
[V1][Sampling] Add opt-in fused generated-token top-logprobs path

### PR DESCRIPTION
**Pull request by yinli-systems**
## What

Adds an opt-in V1 sampler path for generated-token top logprobs:

- `VLLM_ENABLE_FUSED_TOP_LOGPROBS=1` enables the new path.
- The path only applies to `raw_logprobs`, sampled-token logprobs, bounded `num_logprobs`, and no explicit `logprob_token_ids`.
- Unsupported inputs fall back to the existing `log_softmax + topk + gather` implementation.
- Default behavior is unchanged.

The fused CUDA/Triton helper computes the sampled-token logprob, selected-token rank, and top-N logprobs directly from logits without materializing a full `log_softmax(logits)` tensor.

## Why

For generated-token `logprobs`, the current V1 sampler materializes full-vocab logprobs before gathering top-N. On large vocabularies this adds a measurable decode tax even though most of the full matrix is discarded.

This PR is intentionally narrow: it does not change prompt logprobs, full-vocab logprobs, processed-logprobs modes, or explicit token-id logprobs.

## Evidence

External artifact repo: https://github.com/Kevin-Li-2025/single-gpu-inference-lab

A100 vLLM serving A/B, Qwen vocab 152064, `logprobs=20`, median-of-3 paired runs:

- c1: median ITL +0.96%, output tok/s +0.86%, p99 ITL -0.37%, median TTFT -11.43%
- c4: median ITL +1.15%, output tok/s +1.15%, p99 ITL +0.09%, median TTFT +1.04%
- c16: median ITL +1.17%, output tok/s +1.29%, p99 ITL +3.44%, median TTFT -0.90%

Conformance smoke against the native vLLM path:

- text match: true
- token sequence match: true
- token logprobs match: true
- top-logprobs key/value match: true
- max token-logprob diff: 1.1920928955078125e-06
- max top-logprob diff: 1.1920928955078125e-06

Kernel microbenchmark against the PyTorch compatibility path, A100 BF16, vocab 152064, `top_n=20`:

- batch 1: 2.16x
- batch 4: 2.56x
- batch 16: 2.31x

## Tests

- `ruff check vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py`
- `ruff format --check vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py`
- `python3 -m py_compile vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py`
- `git diff --check`

I could not run local pytest in this checkout because the local environment is missing the upstream test dependency `tblib` while importing `tests/conftest.py`.

### CODE REVIEW

**Comment by yinli-systems** [comment]
Maintainer note / self-review:

I re-evaluated this PR against the project value bar. I think it is worth keeping as a draft for review because it targets a narrow serving cost that vLLM currently pays when `logprobs` are requested: full-vocab `log_softmax` materialization followed by top-k/gather, even though the response only needs the sampled-token logprob/rank and top-N entries.

Why the risk is bounded:

- Default behavior is unchanged. The new path is behind `VLLM_ENABLE_FUSED_TOP_LOGPROBS=1`.
- The gate is narrow: V1 sampler, `raw_logprobs`, bounded generated-token `logprobs`, no explicit `logprob_token_ids`, no full-vocab `num_logprobs=-1`.
- Unsupported inputs return `None` and fall back to the existing implementation.
- The raw-logits borrow is only used before in-place sampler processors mutate the float32 logits tensor; fp32 logits are cloned for safety.

Evidence before asking for full CI:

- A100 BF16 kernel microbenchmark, vocab 152064, `top_n=20`: 2.16x-2.56x over the PyTorch compatibility path.
- A100 paired vLLM serving A/B, median-of-3, `logprobs=20`: c1/c4/c16 all show +0.86% to +1.29% output tok/s and +0.96% to +1.17% median ITL improvement.
- Conformance smoke: text, token ids, token logprobs, and top-logprobs key/value outputs match the native path; max observed logprob diff is ~1.2e-6.

Local checks run on this branch:

```bash
ruff check vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py
ruff format --check vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py
python3 -m py_compile vllm/v1/sample/ops/fused_top_logprobs.py vllm/v1/sample/sampler.py tests/v1/sample/test_fused_top_logprobs.py
git diff --check
```

Local pytest is blocked in my checkout by a missing upstream test dependency (`tblib`) while importing `tests/conftest.py`; the added test itself is CPU-safe and only checks the env gate / CPU fallback path.

If this direction looks reasonable, the next useful reviewer action is adding the `ready` or `verified` label so the full pre-commit/CI workflow can run.
